hs_core/hydroshare/utils.py

```diff
@@ -986,6 +978,31 @@ def add_file_to_resource(resource, f, folder=None, source_name='',
     return ret
 
 
+def push_res_to_geohub(url, user, shortkey):
+    # push resource (with uuid as shortkey input) to geohub iRODS zone for geohub tool launch
+    # and return the new tool launch url
+    res = get_resource_by_shortkey(shortkey)
+    istorage = res.get_irods_storage()
+    src_path = res.root_path
+    # delete all temporary resources copied to this user's space before pushing this resource
+    dest_path = os.path.join(settings.GEOHUB_HS_IRODS_PATH, user.username)
+    if istorage.exists(dest_path):
+        istorage.delete(dest_path)
+    dest_path = os.path.join(dest_path, shortkey)
+    istorage.copyFiles(src_path, dest_path, settings.GEOHUB_HS_IRODS_RESC)
+    try:
+        fname = res.get_res_file_name()
+    except Exception as ex:
```

Comment:
along the same lines as my previous comment on [NetcdfResource](https://github.com/hydroshare/hydroshare/pull/2675/files#diff-fb18c7dde2835b95bfb3957cc42d4583).  Is it the correct move to just continue along when the resource file name doesn't exist?